### PR TITLE
Update setup.py

### PR DESCRIPTION
**Pull request by stas00**
I was tweaking m4's `setup.py` with some goodies from here and saw a few minor improvements that could be made here:

- clarify the license and description
- add `jax` to the keywords
- add recent python versions

If anything doesn't fit please let me know and I will remove it.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -420,11 +420,11 @@ def run(self):
     version="4.27.0.dev0",  # expected format is one of x.y.z.dev0, or x.y.z.rc1 or x.y.z (no to dashes, yes to dots)
     author="The Hugging Face team (past and future) with the help of all our contributors (https://github.com/huggingface/transformers/graphs/contributors)",
     author_email="transformers@huggingface.co",
-    description="State-of-the-art Machine Learning for JAX, PyTorch and TensorFlow",
+    description="State-of-the-art Machine Learning framework for JAX, PyTorch and TensorFlow",
```
Comment (sgugger) [inline setup.py:423]: Let's leave this as it is to match the README. Transformers is not a framework, it is a repository of models.

File: setup.py [hunk 2]
```diff
@@ -446,6 +446,8 @@ def run(self):
         "Programming Language :: Python :: 3.7",
         "Programming Language :: Python :: 3.8",
         "Programming Language :: Python :: 3.9",
+        "Programming Language :: Python :: 3.10",
+        "Programming Language :: Python :: 3.11",
```
Comment (sgugger) [inline setup.py:450]: I'm not sure this should be added now as TensorFlow does not support python 3.11 yet, and I'm not too sure for PyTorch.

**Review by sgugger — COMMENTED** [review]
Thanks for your PR! Left a few comments.

File: setup.py (hunk 2 above)
Comment (stas00) [reply setup.py:450]: that's interesting, that means I was under a wrong impression that python is backward compatible.

is it ok to keep 3.10?

File: setup.py (hunk 1 above)
Comment (stas00) [reply setup.py]: oh, for sure. It just feels like a noun is missing there when I read "Machine Learning <something>"

but I removed it as you suggested.

File: setup.py (hunk 2 above)
Comment (sgugger) [reply setup.py]: Yes 3.10 is definitely supported!

**Review by sgugger — APPROVED** [review]

**Pull request merged**